devicetypes/smartthings/zigbee-multi-button.src/zigbee-multi-button.groovy

```diff
@@ -0,0 +1,208 @@
+		getButtonResult("release", button)
+	}
+}
+
+def getButtonResult(buttonState, buttonNumber = 1) {
+	def event = [:]
+	if (buttonState == 'release') {
+		def timeDiff = now() - state.pressTime
+		if (timeDiff > 10000) {
+			return event
+		} else {
+			buttonState = timeDiff < holdTime ? "pushed" : "held"
+			def descriptionText = "${device.displayName[0..-2]} button $buttonNumber was $buttonState"
+			event = createEvent(name: "button", value: buttonState, data: [buttonNumber: buttonNumber], descriptionText: descriptionText, isStateChange: true)
+			if(buttonNumber != 1) {
+				sendEventToChild(buttonNumber, event)
+				return createEvent(descriptionText: descriptionText)
+			} else {
+				return event
+			}
+		}
+	} else if (buttonState == 'press') {
+		state.pressTime = now()
+		return event
+	}
+}
+
+def sendEventToChild(buttonNumber, event) {
+	String childDni = "${device.deviceNetworkId}:$buttonNumber"
+	def child = childDevices.find { it.deviceNetworkId == childDni }
+	child?.sendEvent(event)
+}
+
+def getBatteryPercentageResult(rawValue) {
+	log.debug 'Battery'
+	def volts = rawValue / 10
+	if (volts > 3.0 || volts == 0 || rawValue == 0xFF) {
+		[:]
+	} else {
+		def result = [
+				name: 'battery'
+		]
+		def minVolts = 2.1
+		def maxVolts = 3.0
+		def pct = (volts - minVolts) / (maxVolts - minVolts)
+		result.value = Math.min(100, (int)(pct * 100))
+		def linkText = getLinkText(device)
+		result.descriptionText = "${linkText} battery was ${result.value}%"
+		createEvent(result)
+	}
+}
+
+def refresh() {
+	return zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, batteryVoltage) +
+			zigbee.enrollResponse()
+}
+
+def configure() {
+	def bindings = getModelBindings(device.getDataValue("model"))
+	return zigbee.onOffConfig() + zigbee.levelConfig() +
+			zigbee.configureReporting(zigbee.POWER_CONFIGURATION_CLUSTER, batteryVoltage, DataType.UINT8, 30, 21600, 0x01) +
+			zigbee.enrollResponse() +
+			zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, batteryVoltage) + bindings
+}
+
+def installed() {
+	initialize()
+	if(childDevices) {
+		def event
+		for(def endpoint : 2..device.currentValue("numberOfButtons")) {
+			event = createEvent(name: "button", value: "pushed", isStateChange: true)
+			sendEventToChild(endpoint, event)
+		}
+	}
+}
+
+def updated() {
+	initialize()
+}
+
+def initialize() {
+	def numberOfButtons = modelNumberOfButtons[device.getDataValue("model")]
+	sendEvent(name: "numberOfButtons", value: numberOfButtons, displayed: false)
+	sendEvent(name: "DeviceWatch-Enroll", value: JsonOutput.toJson([protocol: "zigbee", scheme:"untracked"]), displayed: false)
+	if(!childDevices) {
+		addChildButtons(numberOfButtons)
+	}
+}
+
+private addChildButtons(numberOfButtons) {
+	for(def endpoint : 2..numberOfButtons) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = device.displayName[0..-2] + "${endpoint}"
```

Comment:
Same note as above for the displayName.

Reply:
Like with the other device handlers (or at least Z-Wave Dual Switch which I just checked), I would add a check for `device.displayName.endsWith('1')`.

I'd use:

`def componentLabel = (device.displayName.endsWith('1') ? device.displayName[0..-2] : device.displayName) + "${endpoint}"`. It might be better to do `.endsWith(' 1')` with the space since you are trimming two characters.

Reply:
Done.